sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/CustomModelTransforms.java

```diff
@@ -0,0 +1,118 @@
+import com.azure.ai.formrecognizer.models.FormRecognizerError;
+import com.azure.ai.formrecognizer.models.ModelTrainingStatus;
+import com.azure.ai.formrecognizer.models.TrainingDocumentInfo;
+import com.azure.ai.formrecognizer.models.TrainingStatus;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to convert service level custom form related models to SDK exposed models.
+ */
+final class CustomModelTransforms {
+    public static final float DEFAULT_CONFIDENCE_VALUE = 1.0f;
+    private static final ClientLogger LOGGER = new ClientLogger(CustomModelTransforms.class);
+
+    private CustomModelTransforms() {
+    }
+
+    /**
+     * Helper method to convert the {@link Model model Response} from service to {@link CustomFormModel}.
+     *
+     * @param modelResponse The {@code Model model response} returned from the service.
+     *
+     * @return The {@link CustomFormModel}.
+     */
+    static CustomFormModel toCustomFormModel(Model modelResponse) {
+        ModelInfo modelInfo = modelResponse.getModelInfo();
+        if (modelInfo.getStatus() == ModelStatus.INVALID) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException(String.format("Model Id %s returned with status: %s",
+                    modelInfo.getModelId(), modelInfo.getStatus())));
+        }
+
+        List<TrainingDocumentInfo> trainingDocumentInfoList =
+            modelResponse.getTrainResult().getTrainingDocuments().stream()
+                .map(trainingDocumentItem -> new TrainingDocumentInfo(
+                    trainingDocumentItem.getDocumentName(),
+                    TrainingStatus.fromString(trainingDocumentItem.getStatus().toString()),
+                    trainingDocumentItem.getPages(),
+                    new IterableStream<FormRecognizerError>(transformTrainingErrors(trainingDocumentItem.getErrors()))))
+                .collect(Collectors.toList());
+
+        List<CustomFormSubModel> subModelList = new ArrayList<>();
+        String formType = "form-";
+        // unlabeled model
+        if (modelResponse.getKeys() != null) {
+            Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
+            modelResponse.getKeys().getClusters().forEach((clusterKey, clusterFields) -> {
+                for (int i = 0, clusterFieldsSize = clusterFields.size(); i < clusterFieldsSize; i++) {
+                    String eachField = clusterFields.get(i);
+                    String fieldLabel = "field-" + i;
+                    fieldMap.put(fieldLabel, new CustomFormModelField(fieldLabel, eachField, null));
+                }
+                subModelList.add(new CustomFormSubModel(
+                    DEFAULT_CONFIDENCE_VALUE,
```

Comment:
Will this apply even if this is accuracy and not confidence?

Reply:
It shouldn't apply to accuracy, updated!